### DISCUSSION TITLE
What scroll plugin is being used on this site?

### DISCUSSION THREAD

**Discussion by RotasArepo** [vercel/next.js#22279] [Help] [answered]
I understand this is certainly an unconventional discussion, but I just had a quick question. I've been looking for a scroll-based animation library for a long, long time now. On your website, [on this page](https://nextjs.org/commerce), you have that scroll animation with the storefront image. That animation is unbelievably smooth.

Does anyone know what library Next is using to achieve this? Thank you all so much.

**Comment by michrome** [comment, marked as answer]
It's smooth because it's using a scroll event listener in conjunction with `requestAnimationFrame` and `translate3d`. There's a great [example of how to do something like on the MDN site](https://developer.mozilla.org/en-US/docs/Web/API/Document/scroll_event#scroll_event_throttling).

Their equivalent of `doSomething(scrollPos)` is updating the image's `transform` CSS property with a [`translate3d() CSS function`](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/translate3d()) that's being passed values based on the scroll position.

```js
"div",
  {
    id: "display-src",
    className: b().source,
    children: (0, n.jsx)("img", {
      src: "/static/images/commerce/ecommerce.png",
      alt: "Ecommerce App",
      style: l,
    }),
  };
```
…
```js
l = {
  transform: "translate3d(0,calc("
    .concat(((e = c), e * (2 - e) * r), "px + 3.8%),0) scale(")
    .concat(Math.min(0.94 + 0.16 * c, k), ") rotateX(")
    .concat(69 * (c - 1), "deg)"),
};
```

To answer your question: I'm not sure where they're relying on a library to do this 😀

**Reply by RotasArepo** [reply]
Thank you so much for your detailed response! I greatly appreciate it!
